drivers/amazonec2/amazonec2.go

```diff
@@ -651,7 +648,9 @@ func (d *Driver) Create() error {
 
 	if err := d.innerCreate(); err != nil {
 		// cleanup partially created resources
-		d.Remove()
+		if err := d.Remove(); err != nil {
+			return err
+		}
```

Comment:
If there was an error during create, then we should also report that. It seems reasonable to me that if there was an error during create, then there would also be an error during remove.